### DISCUSSION TITLE
How do I fix Auth Error in Function logs?

### DISCUSSION THREAD

**Discussion by ChrisB007** [nextauthjs/next-auth#2887] [Help]
Hello friends, 
I am using next-auth for authentication with Github as a provider. Upon deploying to Vercel I am getting this error in my function logs (Vercel):

```
[GET] /api/auth/callback/github?code=a5277c6916f7018a75ac&state=52cd626fc315ecf3cc717bd3eda146ac74acbd6d8c63c15777048f97ae6af4fd
10:57:17:45
2021-10-04T14:57:27.483Z a2961a30-d43f-4832-8b8a-f13d0d5f1861 Task timed out after 10.01 seconds

```
<img width="1279" alt="Screen Shot 2021-10-04 at 11 08 23 AM" src="https://user-images.githubusercontent.com/33075219/135876322-38dc3d3b-dd41-4072-ba1f-eca236082639.png">

<img width="610" alt="Screen Shot 2021-10-04 at 11 05 55 AM" src="https://user-images.githubusercontent.com/33075219/135876013-c39ccb3b-babf-4484-9404-9550825d5309.png">

How can I resolve this?

**Comment by balazsorban44** [comment]
Is this consistent, or random? Maybe a cold function that doesn't respond in time. Not sure if we can do something directly, but if you use getSession in getServerSideProps or in API routes, a possible remedy is on the way, check #1535 for more info

**Reply by ChrisB007** [reply]
Thank you

**Reply by ChrisB007** [reply]
@balazsorban44 - Good day to you. I came to you once with an issue and you helped me. I am coming to you again with another issue that I hope you can help me with. I have a site that I am using NextAuth with for authentication. The login page and redirect works only on my computer's Chrome and in Incognito mode, but when I switch browser to firefox, safari, or via mobile it shows "304 Error".
Please see the screenshot here (https://share.vidyard.com/watch/jjPCXoJKVXPEfjg5K83K8w?). How can  I debug/ fix this? The site is at tuberdome.com if you want to recreate the issue. Thank you in advance

**Reply by balazsorban44** [reply]
Hard to guess, please open a separate discussion and add some code.

**Reply by ChrisB007** [reply]
Hello @balazsorban44 . Thanks for your response, and my apologies for taking this long to respond, I have been sick with flu. I will start another discussion as you suggested and add some code. Thanks again.
